### PR TITLE
main_common_test: workaround for libevent TSAN failures.

### PR DESCRIPTION
**Pull request by htuch**
This is needed to unblock https://github.com/envoyproxy/envoy/pull/6196 while
https://github.com/libevent/libevent/pull/741 (and consequently
https://github.com/envoyproxy/envoy/issues/6083) go unresolved.

It's not particularly wonderful, but it will allow us to make progress independent of the libevent
work.

Risk level: Low
Testing: TSAN runs with https://github.com/envoyproxy/envoy/pull/6196.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: test/exe/main_common_test.cc [hunk 1]
```diff
@@ -245,6 +267,7 @@ TEST_P(AdminRequestTest, AdminRequestGetStatsAndQuit) {
 TEST_P(AdminRequestTest, AdminRequestGetStatsAndKill) {
   startEnvoy();
   started_.WaitForNotification();
+  waitForEnvoyRun();
```
Comment (mattklein123) [inline test/exe/main_common_test.cc]: Do we still need `started_` or can we get rid of that?

File: source/exe/main_common.h [hunk 2]
```diff
@@ -84,6 +84,7 @@ class MainCommon {
 public:
   MainCommon(int argc, const char* const* argv);
   bool run() { return base_.run(); }
+  Event::Dispatcher& dispatcher() { return base_.server()->dispatcher(); }
```
Comment (dnoe) [inline source/exe/main_common.h]: If this is only exposed for test purposes, should we make that explicit with name or comment?

File: test/exe/main_common_test.cc (hunk 1 above)
Comment (htuch) [reply test/exe/main_common_test.cc]: It's not needed, but I want to leave it there to make it easy to undo by just deleting all of `waitForEnvoyRun()` later. I will leave a comment.

File: source/exe/main_common.h (hunk 2 above)
Comment (htuch) [reply source/exe/main_common.h]: Sure.

**Pull request merged**